sdk/appconfiguration/app-configuration/samples-dev/snapshot.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary Demonstrates the CRUD operations on the snapshot.
+ * @azsdk-weight 100
+ */
+import { AppConfigurationClient } from "@azure/app-configuration";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+export async function main() {
+  console.log(`Running snapshot sample`);
+
+  // Set the following environment variable or edit the value on the following line.
+  const connectionString = process.env["APPCONFIG_CONNECTION_STRING"] || "<connection string>";
+  const client = new AppConfigurationClient(connectionString);
+
+  const key2 = "Samples:key2";
+  const key1 = "Samples:key1";
+  const snapshot1 = {
+    name: "mySnapshot",
+    filters: [
+      {
+        key: key1,
+      },
+    ],
+  };
+  const snapshot2 = {
+    name: "mySnapshot2",
+    filters: [
+      {
+        key: key2,
+      },
+    ],
+  };
+
+  // creating a new setting
+  console.log(`Adding in new setting ${key1}`);
+  await client.addConfigurationSetting({ key: key1, value: "value1" });
+
+  // creating a new setting
+  console.log(`Adding in new setting ${key2}`);
+  await client.addConfigurationSetting({ key: key2, value: "value2" });
+
+  // creating a new snapshot
+  const newSnapshot = await client.beginCreateSnapshotAndWait(snapshot1);
```

Comment:
I think the sample will read a little easier if the values of `snapshot1` and `snapshot2` are inlined.